insights-notification-schemas-java/src/main/resources/schemas/Action.json

```diff
@@ -0,0 +1,123 @@
+{
+  "type": "object",
+  "title": "Notification message",
+  "comment": "Generated with https://hellosean1025.github.io/json-schema-visual-editor/",
+  "properties": {
+    "version": {
+      "type": "string",
+      "title": "Version of the notification schema",
+      "default": "2.0.0"
+    },
+    "bundle": {
+      "type": "string",
+      "title": "Bundle's name",
+      "description": "Bundle name as used during application registration"
+    },
+    "application": {
+      "type": "string",
+      "title": "Application's name",
+      "description": "Application name as used during application registration"
+    },
+    "event_type": {
+      "type": "string",
+      "title": "Event type name",
+      "description": "Event type name as used during application registration"
+    },
+    "timestamp": {
+      "type": "string",
+      "format": "date-time",
+      "description": "ISO-8601 formatted date (per platform convention) when the message was sent",
```

Comment:
Does the platform convention say if this has to be a UTC date-time? We are parsing a timezone in `LocalDateTimeValidator` but maybe we don't need it.

Reply:
According to the [guidelines](https://consoledot.pages.redhat.com/docs/dev/developer-references/rest/datatypes.html#_dates_and_times)

> All dates MUST be in [ISO-8601 format](https://xml2rfc.tools.ietf.org/public/rfc/html/rfc3339.html#anchor14)[](https://consoledot.pages.redhat.com/docs/dev/developer-references/rest/datatypes.html#_ip_addresses_and_hostnames), and times must be in UTC. Applications that require subsecond resolution MAY use it in both inputs and outputs. Applications requiring sub-second resolution MAY be as precise as necessary, and applications and clients can disregard sub-second information if appropriate.
>
> All times MUST be normalized to UTC in API results. The API must return UTC values, but applications and clients may receive input from users, convert for output, and display times in the user’s local timezone based on preferences.


>  We are parsing a timezone in LocalDateTimeValidator but maybe we don't need it.

Could you elaborate on this?

Reply:
`DateTimeFormatter.ISO_DATE_TIME` (currently used in LocalDateTimeValidator) accepts values such as `2011-12-03T10:15:30+01:00` or `2011-12-03T10:15:30+01:00[Europe/Paris]` which doesn't make sense if all dates and times we receive are supposed to be in UTC.

We could use `DateTimeFormatter.ISO_LOCAL_DATE_TIME`, which only accepts one value format: `2011-12-03T10:15:30`, and assume that all date-time values we receive are in UTC. We would also document that in the schema description.

In any case, we need to know which timezone is used when Jackson deserializes `2011-12-03T10:15:30` which is valid with both `ISO_DATE_TIME` and `ISO_LOCAL_DATE_TIME`. If it uses the system clock, then it's bad and we should enforce UTC.

Reply:
⚠️ Switching from `ISO_DATE_TIME` to `ISO_LOCAL_DATE_TIME` could be a breaking change since `ISO_DATE_TIME` was used with Avro.

Reply:
I'll add some testing around that to clarify

Reply:
Some notes:
  `ISO_LOCAL_DATE_TIME` does not support `2020-07-14T13:22:10Z` - I recall I have seen somee payloads with that format.
  
 I left the original  `ISO_DATE_TIME` but tweaked it a bit so that we reject dates with offset. 

Jackson [deserialize is configured](https://github.com/FasterXML/jackson-datatype-jsr310/blob/master/src/main/java/com/fasterxml/jackson/datatype/jsr310/JavaTimeModule.java#L144) to use `LocalDateTime`